src/x86.rs

```diff
@@ -6,59 +6,16 @@ use core::intrinsics;
 // calling convention which can't be implemented using a normal Rust function
 
 // NOTE These functions are never mangled as they are not tested against compiler-rt
-// and mangling ___chkstk would break the `jmp ___chkstk` instruction in __alloca
 
 intrinsics! {
-    #[naked]
-    #[cfg(all(
-        windows,
-        target_env = "gnu",
-        not(feature = "no-asm")
-    ))]
-    pub unsafe extern "C" fn ___chkstk_ms() {
-        core::arch::asm!(
-            "push   %ecx",
-            "push   %eax",
-            "cmp    $0x1000,%eax",
-            "lea    12(%esp),%ecx",
-            "jb     1f",
-            "2:",
-            "sub    $0x1000,%ecx",
-            "test   %ecx,(%ecx)",
-            "sub    $0x1000,%eax",
-            "cmp    $0x1000,%eax",
-            "ja     2b",
-            "1:",
-            "sub    %eax,%ecx",
-            "test   %ecx,(%ecx)",
-            "pop    %eax",
-            "pop    %ecx",
-            "ret",
-            options(noreturn, att_syntax)
-        );
-    }
-
-    // FIXME: __alloca should be an alias to __chkstk
     #[naked]
     #[cfg(all(
         windows,
         target_env = "gnu",
         not(feature = "no-asm")
     ))]
     pub unsafe extern "C" fn __alloca() {
```

Comment:
Are you sure that this is correct? rustc's name mangling will add another underscore in front of this function, so the final symbol will be `___alloca`.

Reply:
Good point, this should of course be `_alloca`. 🤦‍♂️

Commit d32d5eeeb565c3c6f4288976e4bc07d82d9694fd fixes this.